### ISSUE TITLE
doxygen -x_noenv option to not expand environment variables

### ISSUE THREAD

**Issue by ferdymercury** [doxygen/doxygen#9187]
**Describe the bug**
doxygen -x expands environment variables before doing the 'diff'.

**Expected behavior**
It would be interesting to have a "-x2" option that does not substitute the environment variables but rather treats them as plain text so that the 'real diff' is shown.

**Screenshots**
Let's assume an external user is inspecting a 'foreign Doxyfile' and does not have info about whether there are environmental variables defined within (thus, they are undefined in the shell). He gets:
```
doxygen -x Doxyfile
# Difference with default Doxyfile 1.9.4 (70d1c6baf5107dcfaae6fee02ea6b3ef93cea61c*)
```
So it's empty, no differences. Versus the manual diff I do normally:
```
diff Doxyfile Doxyfile.orig
41c41
< PROJECT_NUMBER         = $(SOME_VARIABLE)
---
> PROJECT_NUMBER         =
```
which shows the difference.

**To Reproduce**
- doxygen -g
- cp Doxyfile Doxyfile.orig
- Open the generated file and set PROJECT_NUMBER           = $(SOME_VARIABLE)
- doxygen -x Doxyfile
- diff Doxyfile Doxyfile.orig

**Version**
Doxyfile 1.9.4 (70d1c6baf5107dcfaae6fee02ea6b3ef93cea61c*)

**Comment by albert-github** [comment]
The idea behind `doxygen -x` is that it gives the differences between the Doxyfile that will be used when running `doxygen` and the default doxygen settings.
The mentioned idea / use case of having the possibility to retain the original environment settings is not a bad idea, though `-x2` probably would be better as `-x_noenv`.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9189

**Comment by ferdymercury** [comment]
Thanks for the fix!

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.4.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
